src/us/kbase/workspace/test/workspace/WorkspaceTest.java

```diff
@@ -2602,31 +2601,11 @@ public void saveObjectWithTypeChecking() throws Exception {
 		
 		
 		Provenance goodids = new Provenance(userfoo);
-		goodids.addAction(new Provenance.ProvenanceAction().withWorkspaceObjects(Arrays.asList("typecheck/1/1")));
+		goodids.addAction(ProvenanceAction.getBuilder()
+				.withWorkspaceObjects(list("typecheck/1/1")).build());
 		data.set(1, new WorkspaceSaveObject(obj3, data3, abstype0, null, goodids, false));
 		ws.saveObjects(userfoo, wspace, data, getIdFactory()); //should work
 		
-		Provenance badids = new Provenance(userfoo);
-		badids.addAction(new Provenance.ProvenanceAction().withWorkspaceObjects(Arrays.asList("foo/bar/baz")));
-		final ObjectIDNoWSNoVer obj4 = new ObjectIDNoWSNoVer("obj4");
-		data.set(1, new WorkspaceSaveObject(obj4, data3, abstype0, null, badids, false));
-		failSave(userfoo, wspace, data, new TypedObjectValidationException(
-				"Object #2, obj4 has unparseable provenance reference foo/bar/baz: Unable to " +
-				"parse version portion of object reference 'foo/bar/baz' to an integer"));
-		
-		badids = new Provenance(userfoo);
-		badids.addAction(new Provenance.ProvenanceAction().withWorkspaceObjects(Arrays.asList((String) null)));
-		data.set(1, new WorkspaceSaveObject(obj4, data3, abstype0, null, badids, false));
-		failSave(userfoo, wspace, data, new TypedObjectValidationException(
-				"Object #2, obj4 has a null provenance reference"));
-		
-		badids = new Provenance(userfoo);
-		badids.addAction(new Provenance.ProvenanceAction().withWorkspaceObjects(Arrays.asList("")));
-		data.set(1, new WorkspaceSaveObject(obj4, data3, abstype0, null, badids, false));
-		failSave(userfoo, wspace, data, new TypedObjectValidationException(
-				"Object #2, obj4 has invalid provenance reference: IDs may not be null or the " +
-				"empty string"));
-		
```

Comment:
These 3 tests are now pointless because bad refs are rejected by the provenance action class and are tested in the unit tests thereof. I added some integration tests in the JsonRPCLayerTest to exercise the error pathway though.